pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -248,6 +261,76 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	return newReconciledNormal(kc.Namespace, kc.Name)
 }
 
+func (r *Reconciler) setupSubscriptionStatusWatchers(ctx context.Context, channel *v1beta1.KafkaChannel) error {
+	var err error
+	if channel.Spec.SubscribableSpec.Subscribers != nil {
+		for _, s := range channel.Spec.SubscribableSpec.Subscribers {
+			groupID := fmt.Sprintf("kafka.%s.%s.%s", channel.Namespace, channel.Name, string(s.UID))
```

Comment:
can you create a function for this and use it in `dispatcher.go` as well?

Reply:
I refactored that part, can you check again?